src/feature-feedback.ts

```diff
@@ -0,0 +1,608 @@
+import {
+  html,
+  css,
+  LitElement,
+  nothing,
+  CSSResultGroup,
+  PropertyValues,
+  TemplateResult,
+} from 'lit';
+import { property, customElement, query, state } from 'lit/decorators.js';
+
+import type {
+  SharedResizeObserverInterface,
+  SharedResizeObserverResizeHandlerInterface,
+} from '@internetarchive/shared-resize-observer';
+import type {
+  RecaptchaManagerInterface,
+  RecaptchaWidgetInterface,
+} from '@internetarchive/recaptcha-manager';
+import type { FeatureFeedbackServiceInterface } from './feature-feedback-service';
+import type { Vote } from './models';
+
+import { thumbsUp } from './img/thumb-up';
+import { thumbsDown } from './img/thumb-down';
+
+@customElement('feature-feedback')
+export class FeatureFeedback
+  extends LitElement
+  implements SharedResizeObserverResizeHandlerInterface
+{
+  @property({ type: String }) featureIdentifier?: string;
+
+  @property({ type: String }) prompt = 'Do you find this feature useful?';
+
+  @property({ type: String }) buttonText = 'Beta';
+
+  @property({ type: Object }) recaptchaManager?: RecaptchaManagerInterface;
+
+  @property({ type: Object }) resizeObserver?: SharedResizeObserverInterface;
```

Comment:
This is where the power of dependency injection comes into play! By using an interface instead of the actual object type, we can substitute the "real" `SharedResizeObserver` with a different implementation of it (ie lets say someone writes a more efficient version) or a mock `SharedResizeObserver` that specifically acts like we want it to during testing. ie:
```ts
// test.ts
export class MockSharedResizeObserver implements SharedResizeObserverInterface {
  // implement the SharedResizeObserverInterface to do whatever you need
}

const mockRO = new MockSharedResizeObserver();
<feature-feedback .resizeObserver=${mockRO}></feature-feedback>
```
The `<feature-feedback>` doesn't care what class it's actually given, as long as it conforms to the `SharedResizeObserverInterface`